WorkflowSwiftUI/Sources/Store.swift

```diff
@@ -0,0 +1,453 @@
+import CasePaths
+import IdentifiedCollections
+import Perception
+import SwiftUI
+import Workflow
+
+/// Provides access to a workflow's state and actions from within an ``ObservableScreen``.
+///
+/// The store wraps an ``ObservableModel`` and provides controlled access to members through dynamic
+/// member lookup:
+/// - state properties
+/// - action sinks
+/// - child stores using nested `ObservableModel`s
+///
+/// Because arbitrary properties on the model cannot be tracked for observation, any other types of
+/// properties will not be accessible.
+///
+/// For state properties that are writable, an automatic `Binding` can be derived by annotating the
+/// store with `@Bindable`. These bindings will use the workflow's state mutation sink.
+///
+/// All properties can be turned into bindings by appending `sending(store:action:)` or
+/// `sending(closure:)` to specify the "write" action. For properties that are already writable,
+/// this will refine the binding to send a custom action instead of the built-in state mutation
+/// sink.
+///
+@dynamicMemberLookup
+public final class Store<Model: ObservableModel>: Perceptible {
+    public typealias State = Model.State
+
+    private var model: Model
+    private let _$observationRegistrar = PerceptionRegistrar()
+
+    private var childStores: [AnyHashable: ChildStore] = [:]
+    private var childModelAccesses: [AnyHashable: ChildModelAccess] = [:]
+    private var invalidated = false
+
+    static func make(model: Model) -> (Store, (Model) -> Void) {
+        let store = Store(model)
+        return (store, store.setModel)
+    }
+
+    fileprivate init(_ model: Model) {
+        self.model = model
+    }
+
+    var state: State {
+        _$observationRegistrar.access(self, keyPath: \.state)
+        return model.accessor.state
+    }
+
+    private func send<Value>(keyPath: WritableKeyPath<State, Value>, value: Value) {
+        guard !invalidated else {
```

Comment:
If this is false, the change will be silently dropped – should this be a fatal instead, or can we add some sort of additional validation?

Reply:
Hmm, I'm not sure we can. 🤔

Besides preventing the cache from growing unbounded, invalidating stores is useful because SwiftUI itself has been known to send values into bindings long after their views are invalid, and we don't want those values (and they will potentially assert at the workflow level). I'm not sure there's a way we could tell if a value came from SwiftUI misbehaving or programmer misbehaving.